backend/typescript/services/implementations/schedulingService.ts

```diff
@@ -148,6 +147,54 @@ class SchedulingService implements ISchedulingService {
     return schedulingDtos;
   }
 
+  async getSchedulingsByVolunteersNeeded(
+    isVolunteerSlotFilled?: boolean,
+  ): Promise<SchedulingDTO[]> {
+    let schedulingDtos: Array<SchedulingDTO> = [];
+    try {
+      const schedulings: Array<Scheduling> = await Scheduling.findAll({
+        where:
+          isVolunteerSlotFilled === undefined
+            ? {
+                volunteer_needed: true,
+              }
+            : {
+                volunteer_needed: true,
+                volunteer_id: isVolunteerSlotFilled ? { [Op.ne]: null } : null,
+              },
+        order: [["start_time", "ASC"]],
+      });
+      schedulingDtos = schedulings.map((scheduling) => {
+        return {
+          id: String(scheduling.id),
+          donorId: String(scheduling.donor_id),
+          categories: scheduling.categories,
+          size: scheduling.size,
+          isPickup: scheduling.is_pickup,
+          pickupLocation: scheduling.pickup_location,
+          dayPart: scheduling.day_part,
+          startTime: scheduling.start_time,
+          endTime: scheduling.end_time,
+          status: scheduling.status,
+          volunteerNeeded: scheduling.volunteer_needed,
+          volunteerTime: scheduling.volunteer_time,
+          frequency: scheduling.frequency,
+          recurringDonationId: String(scheduling.recurring_donation_id),
+          recurringDonationEndDate: scheduling.recurring_donation_end_date,
+          notes: scheduling.notes,
+          volunteerId: String(scheduling.volunteer_id),
+        };
+      });
+    } catch (error) {
+      Logger.error(
+        `Failed to get schedulings. Reason = ${getErrorMessage(error)}`,
```

Comment:
nit: let's write a more descriptive error here that's distinct from the `getSchedulings` function
```suggestion
        `Failed to get schedulings by volunteers needed. Reason = ${getErrorMessage(error)}`,
```